### ISSUE TITLE
navigation requests w/ StaleWhileRevalidate + broadcastUpdate = TypeError: Deferred_mjs.Deferred is not a constructor

### ISSUE THREAD

**Issue by goyavo** [GoogleChrome/workbox#2045]
**Library Affected**:
workbox-broadcast-cache-update

**Browser & Platform**:
Chrome

**Issue or Feature Request Description**:
When trying to use workbox-broadcast-cache-update for navigation requests using the StaleWhileRevalidate for which there is an update (using the Last-Modified header), I get the following error: 

```
Uncaught (in promise) TypeError: Deferred_mjs.Deferred is not a constructor
    at BroadcastCacheUpdate._windowReadyOrTimeout (BroadcastCacheUpdate.mjs:154)
    at sendUpdate (BroadcastCacheUpdate.mjs:106)
    at BroadcastCacheUpdate.notifyIfUpdated (BroadcastCacheUpdate.mjs:112)
    at Plugin.cacheDidUpdate (Plugin.mjs:80)
    at Object.putWrapper [as put] (cacheWrapper.mjs:99)
```

I am already using workbox-broadcast-cache-update with native Fetch API requests on loaded pages and it's working great. Something to do here with the page not being loaded? Is there a workaround or is there is no way to use BroadcastCacheUpdate for navigation requests?

**Comment by jeffposnick** [comment]
It looks like it's this line that triggers the error:

https://github.com/GoogleChrome/workbox/blob/7c873eb2b7e4530fb554c41d9c425c5167e45de7/packages/workbox-broadcast-update/BroadcastCacheUpdate.mjs#L169

@philipwalton, is this potentially due to some class name mangling introduced by minification?

**Comment by philipwalton** [comment]
Hmmm, it looks like the `Deferred` object is never exported from `workbox-core/_private.mjs`: https://github.com/GoogleChrome/workbox/blob/v4.3.0/packages/workbox-core/_private.mjs#L22-L32

And this passes our unit tests because we do a custom build for those, so I guess we don't cover this code path in our integrations tests.

We should probably also separately have a test to ensure `_private` symbols are properly exported, as this is sort of one-off logic in our build.

**Comment by philipwalton** [comment]
@goyavo the fix for this is in [v4.3.1](https://github.com/GoogleChrome/workbox/releases/tag/v4.3.1).
